src/main/java/picard/sam/SamToFastqWithTags.java

```diff
@@ -0,0 +1,225 @@
+        return generateTagWriters(readGroups, factory);
+    }
+
+    // generate writers
+    private Map<SAMReadGroupRecord, List<FastqWriter>> generateTagWriters(final List<SAMReadGroupRecord> samReadGroupRecords,
+                                                                          final FastqWriterFactory factory) {
+        final Map<SAMReadGroupRecord, List<FastqWriter>> writerMap = new HashMap<>();
+
+        if (!OUTPUT_PER_RG) {
+            /* Prepare tag writers based on sequence tag groups provided in command line */
+
+            final List<FastqWriter> tagFastqWriters = makeTagWriters(null).stream().map(factory::newWriter).collect(Collectors.toList());
+
+            writerMap.put(null, tagFastqWriters);
+            for (final SAMReadGroupRecord rg : samReadGroupRecords) {
+                writerMap.put(rg, tagFastqWriters);
+            }
+        } else {
+            /* prepare tag writers based on readgroup names */
+            for (final SAMReadGroupRecord rg : samReadGroupRecords) {
+                List<FastqWriter> tagWriters = makeTagWriters(rg).stream().map(factory::newWriter).collect(Collectors.toList());
+
+                writerMap.put(rg, tagWriters);
+            }
+        }
+        return writerMap;
+    }
+
+    // Creates fastq writers based on readgroup passed in and sequence tag groupings from command line
+    private List<File> makeTagWriters(final SAMReadGroupRecord readGroup) {
+        String baseFilename = null;
+        if (readGroup != null) {
+            if (RG_TAG.equalsIgnoreCase("PU")) {
+                baseFilename = readGroup.getPlatformUnit() + "_";
+            } else if (RG_TAG.equalsIgnoreCase("ID")) {
+                baseFilename = readGroup.getReadGroupId() + "_";
+            }
+            if (baseFilename == null) {
+                throw new PicardException("The selected RG_TAG: " + RG_TAG + " is not present in the bam header.");
+            }
+        } else {
+            baseFilename = "";
+        }
+
+        List<File> tagFiles = new ArrayList<>();
+        for (String tagSplit : SEQUENCE_TAG_GROUP) {
+            String fileName = baseFilename + tagSplit.replace(",", "_");
+            fileName = IOUtil.makeFileNameSafe(fileName);
+            fileName += COMPRESS_OUTPUTS_PER_TAG_GROUP ? ".fastq.gz" : ".fastq";
+
+            final File result = (OUTPUT_DIR != null)
+                    ? new File(OUTPUT_DIR, fileName)
+                    : new File(FASTQ.getParent(), fileName);
+            IOUtil.assertFileIsWritable(result);
+            tagFiles.add(result);
+        }
+        return tagFiles;
+    }
+
+    // Setting up the Groupings of Sequence Tags, Quality Tags, and Separator Strings so we dont have to calculate them for every loop
+    private void setupTagSplitValues() {
+        if (SEQUENCE_TAG_GROUP.isEmpty()) return;
```

Comment:
please split across lines and add `{` `}`

Reply:
I can actually get rid of this `if` entirely, thanks.